src/Solution/ParticleTracker/Particle.f90

```diff
@@ -40,12 +40,15 @@ module ParticleModule
     ! stop criteria
     integer(I4B), public :: istopweaksink !< weak sink option (0: do not stop, 1: stop)
     integer(I4B), public :: istopzone !< stop zone number
+    integer(I4B), public :: idrymeth !< dry tracking method
     ! state
-    integer(I4B), allocatable, public :: idomain(:) !< tracking domain hierarchy
+    integer(I4B), allocatable, public :: idomain(:) !< tracking domain hierarchy ! TODO: rename to itdomain? idomain
     integer(I4B), allocatable, public :: iboundary(:) !< tracking domain boundaries
-    integer(I4B), public :: icu !< user cell (node) number
+    integer(I4B), public :: icp !< previous cell number (reduced)
+    integer(I4B), public :: icu !< user cell number
     integer(I4B), public :: ilay !< grid layer
-    integer(I4B), public :: izone !< zone number
+    integer(I4B), public :: izone !< current zone number
+    integer(I4B), public :: izp !< previous zone number
```

Comment:
this was the simplest way to achieve some very minimal path-awareness for the particle (needed so we can detect backtracking / prevent infinite loops e.g. at the bottom of wells) &mdash; if we find it necessary to look backwards in more detail or depth in the future, maybe this deserves a separate, cleaner implementation

Reply:
can discuss this in context of backtracking in general

Reply:
just wanted to document that we discussed the backtracking check, and you correctly pointed out that a particle shouldn't be able to immediately return to the cell it came from (on the same time step) because the (normal) flow is in one direction all along a face.  as long as the check prevents backtracking across the same face, that should take care of the issue